src/server/active_tran_server.cpp

```diff
@@ -149,19 +174,39 @@ active_tran_server::init_page_server_hosts (const char *db_name)
       er_clear ();
     }
   exit_code = NO_ERROR;
+
+  // use config to connect
+  //
+  int valid_connection_count = 0;
   for (const cubcomm::node &node : m_connection_list)
     {
-      exit_code = connect_to_page_server (node, db_name);
-      if (exit_code == NO_ERROR)
+      const int local_exit_code = connect_to_page_server (node, db_name);
+      if (local_exit_code == NO_ERROR)
 	{
-	  //found valid host clear the errors rom the bad ones
+	  // found valid host clear the errors from the bad ones
+	  ++valid_connection_count;
+	  exit_code = NO_ERROR;
 	  er_clear ();
 	  // successfully connected to a page server. stop now.
-	  return exit_code;
+	  break;
+	}
+      else
+	{
+	  exit_code = local_exit_code;
 	}
     }
-  // failed to connect to any page server
-  assert (exit_code != NO_ERROR);
+
+  // validate connections vs. config
+  //
+  if (valid_connection_count == 0 && m_has_remote_storage)
+    {
+      assert (exit_code != NO_ERROR);
+      // TODO: Inconsistent setting: remote storage is been specified, but no connection to page servers possible
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_INCONSISTENT_REMOTE_STORAGE_NO_CONN, 0);
```

Comment:
- TODO comment does not seem necessary
- The problem qualifies as error, not warning.
```suggestion
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_INCONSISTENT_REMOTE_STORAGE_NO_CONN, 0);
```

Reply:
Done